### PR TITLE
[CORE-2308] Add pipeline metadata

### PR DESCRIPTION
**Pull request by robert-uhl**

### CODE REVIEW

**Review by jrockway — APPROVED** [review]
Looks good.  Maybe add a simple test for ppsdb?

File: src/pps/pps.proto [hunk 1]
```diff
@@ -1052,6 +1053,19 @@ message PipelinesSummary {
   int64 unhealthy_pipelines = 5;
 }
 
+// PipelinePicker is used to select a single pipeline.  It is a pfs.RepoPicker
+// where the repo type defaults to "spec."
+
+message PipelinePicker {
+  message RepoName {
+    pfs_v2.ProjectPicker project = 1;
+    string name = 2;
```
Comment (jrockway) [inline src/pps/pps.proto:1062]: Should we have a validation rule to require this to be non-empty?

Comment (robert-uhl) [reply src/pps/pps.proto:1062]: The PFS Repo Picker message this copies doesn’t have a validation rule. Maybe it should? OTOH, an invalid picker will get an error when actually used.

**Review by emmajsadams — APPROVED** [review]
lgtm!

**Pull request merged**